harness/determined/pytorch/_lr_scheduler.py

```diff
@@ -46,76 +49,64 @@ def __init__(self, scheduler: torch.optim.lr_scheduler._LRScheduler, step_mode:
-        check.check_eq(
+        self._step_count = self.scheduler._step_count  # type: ignore
+
+    def step(self, *args: Any, **kwargs: Any) -> None:
+        """
+        step is a function for users to call step() on the original LRScheduler
+        if it is in MANUAL_STEP mode.
+        """
+        check.eq(
             self.step_mode,
             LRScheduler.StepMode.MANUAL_STEP,
             "Please use the MANUAL_STEP step mode to call step() on the scheduler.",
         )
-        return self.scheduler.step(*args, **kwargs)
-
-    def get_last_lr(self) -> typing.List:
-        """Return last computed learning rate by current scheduler.
-
-        This function is equivalent to calling get_last_lr() on the wrapped LRScheduler.
-        """
-
-        return self.scheduler.get_last_lr()  # type: ignore
+        self.scheduler.step(*args, **kwargs)
 
-
-class _LRHelper:
-    def __init__(self, lr_scheduler: typing.Optional[LRScheduler]):
-        self._lr_scheduler = None
-        if lr_scheduler:
-            check.check_type(
-                lr_scheduler,
-                LRScheduler,
-                "`create_lr_scheduler` must return a `det.pytorch.LRScheduler`",
-            )
-            self._lr_scheduler = lr_scheduler
-            self._lr_scheduler_count = lr_scheduler.scheduler._step_count  # type: ignore
-
-    def __bool__(self) -> bool:
-        return self._lr_scheduler is not None
-
-    def should_step_lr(
+    def _should_call_step(
         self, batches_completed: int, epoch_length: int, aggregation_frequency: int
     ) -> bool:
-        if self._lr_scheduler:
-            if self._lr_scheduler.step_mode == LRScheduler.StepMode.STEP_EVERY_BATCH:
+        if self.step_mode == LRScheduler.StepMode.STEP_EVERY_BATCH:
+            return True
+        elif self.step_mode == LRScheduler.StepMode.STEP_EVERY_EPOCH:
+            mod = batches_completed % epoch_length
+            if mod == 0 or mod < aggregation_frequency:
                 return True
-            elif self._lr_scheduler.step_mode == LRScheduler.StepMode.STEP_EVERY_EPOCH:
-                mod = batches_completed % epoch_length
-                if mod == 0 or mod < aggregation_frequency:
-                    return True
         return False
 
-    def step(self) -> None:
-        check.check_eq(
-            self._lr_scheduler.scheduler._step_count,  # type: ignore
-            self._lr_scheduler_count,
-            "You cannot call `scheduler.step()` if you have configured "
+    def _call_step(self, *args: Any, **kwargs: Any) -> None:
+        """
+        _call_step is a function for internal training loop to call step() on the
+        original LRScheduler if it is not in MANUAL_STEP mode.
+        """
+        check.not_eq(
+            self.step_mode,
+            LRScheduler.StepMode.MANUAL_STEP,
+            "Please use the MANUAL_STEP step mode to call step() on the scheduler.",
+        )
```

Comment:
It seems like this is either not a necessary check or we need to change the wording on it.